homework01/test/examples/Main.hs

```diff
@@ -1,15 +1,44 @@
-module Main
-  ( main
-  )
-  where
-
--- doctest
-import qualified Test.DocTest as DocTest
+module Main where
 
+import Basis
+import Test.Hspec
 
 main :: IO ()
-main =
-  DocTest.doctest
-    [ "-isrc"
-    , "src/Basis.hs"
-    ]
+main = hspec $ do
+
+  describe "toDigits" $ do
+    it "converts positive integers to a list of digits" $
+      toDigits 1234 `shouldBe` [1, 2, 3, 4]
+
+    it "returns an empty list for 0" $
+      toDigits 0 `shouldBe` []
+
+    it "returns an empty list for negative inputs" $
+      toDigits (-17) `shouldBe` []
+
+  describe "toDigitsRev" $
+    it "converts positive integers to a list of digits reversed" $
+      toDigitsRev 1234 `shouldBe` [4, 3, 2, 1]
+
+  describe "doubleEveryOther" $ do
+    it "doubles every other number beginning from the right (even number of elements)" $
+      doubleEveryOther [8, 7, 6, 5] `shouldBe` [16, 7, 12, 5]
+    
+    it "doubles every other number beginning from the right (odd number of elements)" $
+      doubleEveryOther [1, 2, 3] `shouldBe` [1, 4, 3]
+
+  describe "sumDigits" $
+    it "calculates the sum of sums of the digits of each element in a list" $
+      sumDigits [16, 7, 12, 5] `shouldBe` 22
+
+  describe "validate" $ do
+    it "returns True if an Integer is a valid credit card number" $
+      validate 4012888888881881 `shouldBe` True
+    
+    it "returns True if an Integer is a invalid credit card number" $
+      validate 4012888888881882 `shouldBe` False
+
+  describe "hanoi" $
+    it "gets a list of moves of the stack of discs from first peg to second" $
+      hanoi 2 "a" "b" "c" `shouldBe` [("a", "c"), ("a", "b"), ("c", "b")]
```

Comment:
I think it would be better to increase the disc number to check it works to output more  complex solutions